pkg/controllers/osc/resources/operating_system_config.go

```diff
@@ -180,6 +171,11 @@ func GenerateOperatingSystemConfig(
 		BootstrapKubeconfigSecretName: bootstrapKubeconfigSecretName,
 	}
 
+	var inTreeCCM bool
+	if !externalCloudProvider {
+		inTreeCCM = true
```

Comment:
Not having external CCM enabled doesn't mean in-tree CCM is available. We should still keep the old function, but use it only to determine if a provider has in-tree cloud provider or not. External cloud provider should be determined as in this PR -- depending on the flag value.

Reply:
Oh yeah that makes sense, but maybe change it if the external is false.

Reply:
Yes, that makes sense.

Reply:
At the moment only GCP has no support for external and only has support for internal

Reply:
I agree that we should remove the `external` bit. Having it as a special case just for `kubevirt` is wrong. 

But in general, wouldn't this function still be useful for checking availability for in-tree CCM? 

For example, we use the value from that func here as `InTreeCCMAvailable` in our OSPs:

```
                {{- if .ExternalCloudProvider }}
                --cloud-provider=external \
                {{- else if .InTreeCCMAvailable }}
                --cloud-provider={{- .CloudProviderName }} \
                --cloud-config=/etc/kubernetes/cloud-config \
                {{- end }}
```

Reply:
I would say this should becoming from outside, so the user decides that, not enforce it. So the value should be coming from the outside, propagated to OSM via KKP. So users then decides what to use. However, this should be started all the way from KKP addons. Not just for OSM. In other words, in the cluster object this should have InTreeOrOOT(the name is just an example obviously). Do we have this actually ?

Reply:
We do have a way to determine this in KubeOne https://github.com/kubermatic/kubeone/blob/main/pkg/apis/kubeone/helpers.go#L279, which is similar to how we do this in MC and OSM. For KKP we don't have this but maybe it's worth it just copy this over to KKP and abstract this in MC and OSM both.